src/hotspot/share/opto/addnode.cpp

```diff
@@ -263,66 +263,69 @@ AddNode* AddNode::make(Node* in1, Node* in2, BasicType bt) {
 
 //=============================================================================
 //------------------------------Idealize---------------------------------------
-Node *AddINode::Ideal(PhaseGVN *phase, bool can_reshape) {
+Node* AddNode::IdealIL(PhaseGVN* phase, bool can_reshape, BasicType bt) {
   Node* in1 = in(1);
   Node* in2 = in(2);
   int op1 = in1->Opcode();
   int op2 = in2->Opcode();
   // Fold (con1-x)+con2 into (con1+con2)-x
-  if ( op1 == Op_AddI && op2 == Op_SubI ) {
+  if (op1 == Op_Add(bt) && op2 == Op_Sub(bt)) {
     // Swap edges to try optimizations below
     in1 = in2;
     in2 = in(1);
     op1 = op2;
     op2 = in2->Opcode();
   }
-  if( op1 == Op_SubI ) {
-    const Type *t_sub1 = phase->type( in1->in(1) );
-    const Type *t_2    = phase->type( in2        );
-    if( t_sub1->singleton() && t_2->singleton() && t_sub1 != Type::TOP && t_2 != Type::TOP )
-      return new SubINode(phase->makecon( add_ring( t_sub1, t_2 ) ), in1->in(2) );
+  if (op1 == Op_Sub(bt)) {
+    const Type *t_sub1 = phase->type(in1->in(1));
```

Comment:
I'm not very clear about the current code style which we should follow. Shall we need to align the style in the changed code?

Reply:
Are you commenting about the change from:
if( op1 to if (op1
?
The first style is used in a some places and the guideline is to switch to the second one.

Reply:
How about the style like `const Type *t_sub1`? Whether it should be `const Type* t_sub1`.

Reply:
It should be const Type* t_sub1. I will fix it. Thanks.